oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcClientRegistrationEndpointFilterTests.java

```diff
@@ -0,0 +1,405 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for {@link OidcClientRegistrationEndpointFilter}
+ *
+ * @author Ovidiu Popa
+ * @since 0.0.4
+ */
+public class OidcClientRegistrationEndpointFilterTests {
+
+	private static final OidcClientRegistration.Builder OIDC_CLIENT_REGISTRATION = OidcClientRegistration.builder()
+			.redirectUri("https://localhost:8080/client")
+			.responseType("code")
+			.grantType("authorization_code")
+			.tokenEndpointAuthMethod("basic")
+			.scope("test");
+	private final HttpMessageConverter<OAuth2Error> errorHttpResponseConverter =
+			new OAuth2ErrorHttpMessageConverter();
+	private static RegisteredClientRepository registeredClientRepository;
+	private static OAuth2AuthorizationService authorizationService;
+
+	@BeforeClass
+	public static void init() {
+		registeredClientRepository = mock(RegisteredClientRepository.class);
+		authorizationService = mock(OAuth2AuthorizationService.class);
+	}
+
+	@Before
+	public void setup() {
+		reset(registeredClientRepository);
+		reset(authorizationService);
+	}
+
+	@After
+	public void tearDown() {
+		SecurityContextHolder.clearContext();
+	}
+
+	@Test
+	public void constructorWhenRegisteredClientRepositoryNullThenThrowIllegalArgumentException() {
+		assertThatThrownBy(() -> new OidcClientRegistrationEndpointFilter(null,
+				authorizationService, "/connect/register"))
+				.isInstanceOf(IllegalArgumentException.class)
+				.hasMessage("registeredClientRepository cannot be null");
+	}
+
+	@Test
+	public void constructorWhenAuthorizationServiceNullThenThrowIllegalArgumentException() {
+		assertThatThrownBy(() -> new OidcClientRegistrationEndpointFilter(registeredClientRepository,
+				null, "/connect/register"))
+				.isInstanceOf(IllegalArgumentException.class)
+				.hasMessage("authorizationService cannot be null");
+	}
+
+	@Test
+	public void constructorWhenOidcClientRegistrationUriNullThenThrowIllegalArgumentException() {
+		assertThatThrownBy(() -> new OidcClientRegistrationEndpointFilter(registeredClientRepository, authorizationService, null))
+				.isInstanceOf(IllegalArgumentException.class)
+				.hasMessage("oidcClientRegistrationUri cannot be empty");
+	}
+
+	@Test
+	public void constructorWhenOidcClientRegistrationUriEmptyThenThrowIllegalArgumentException() {
+		assertThatThrownBy(() -> new OidcClientRegistrationEndpointFilter(registeredClientRepository, authorizationService, ""))
+				.isInstanceOf(IllegalArgumentException.class)
+				.hasMessage("oidcClientRegistrationUri cannot be empty");
+	}
+
+	@Test
+	public void doFilterWhenNotConfigurationRequestThenNotProcessed() throws Exception {
```

Comment:
NotConfigurationRequest -> NotRegistrationRequest

Please review all test method names as there are others that need updating